content/en/blog/2024/inpod-traffic-redirection-ambient/index.md

```diff
@@ -0,0 +1,240 @@
+redirecting pod traffic in the host network namespace, as we were during alpha development, was not going to meet our requirements.
+It was evident that achieving our goals in a generic manner across these diverse environments was not possible in this approach.
+
+The fundamental problem with this approach (redirecting traffic in the host network namespace) is that this is precisely the same spot where
+the cluster's primary CNI implementation *must* configure traffic routing/networking rules. This created inevitable conflicts, most critically:
+
+- The primary CNI implementation's basic host-level networking configuration could interfere with the host-level ambient networking configuration from Istio's CNI extension, causing traffic disruption and other conflicts.
+- If users deployed a network policy to be enforced by the primary CNI implementation, the network policy might not be enforced when the
+Istio CNI extension is deployed (depending on how the primary CNI implementation enforces NetworkPolicy)
+
+While we could design around this on a case-by-case basis for _some_ primary CNI implementations, we could not sustainably approach
+universal CNI support. We considered eBPF, but realized any eBPF implementation would have the same basic problem, as there is no
+standardized way to safely chain/extend arbitrary eBPF programs at this time, and we would still potentially have a hard time supporting
+non-eBPF CNIs with this approach.
+
+### Addressing the challenges
+
+A new solution was necessary - doing redirection of any sort in the node's network namespace would create unavoidable conflicts,
+unless we compromised our compatibility requirements.
+
+In sidecar mode, it is trivial to configure traffic redirection between the sidecar and application pod, as both operate within
+the pod's network namespace. This led to a light-bulb moment: why not mimic sidecars, and configure the redirection in
+the application pod's network namespace?
+
+While this sounds like a "crazy simple" thought, is this even possible, given ztunnel runs
+in the Istio system namespace? After some research, we discovered a Linux process running in one network namespace
+could create and own listening sockets within another network namespace, which is a basic Linux socket capability.
+However, to make this work, and cover all pod lifecycle scenarios we had to make architectural changes to the ztunnel
+as well as the `istio-cni` agent.
+
+After prototyping and sufficiently validating that this innovative approach does work for all the Kubernetes platforms we have
+access to, we built confidence in the work and decided to contribute this new traffic redirection
+model. An *in-Pod* traffic redirection mechanism between workload pods and the ztunnel node proxy component that has
+been built from the ground up to be highly compatible with all major cloud providers and CNIs.
```

Comment:
```suggestion
After prototyping and sufficiently validating that this novel approach does work for all the Kubernetes platforms we have access to, we built confidence in the work and decided to contribute this new traffic redirection
model. An *in-Pod* traffic redirection mechanism between workload pods and the ztunnel node proxy component that has been built from the ground up to be highly compatible with all major cloud providers and CNIs.
```

We said `innovative` once, let's switch it up and say `novel` here, to avoid repetition.